docs/words/examples/cursesclient.py

```diff
@@ -182,12 +182,18 @@ def close(self):
         curses.echo()
         curses.endwin()
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+
 if __name__ == '__main__':
     stdscr = curses.initscr() # initialize curses
-    screen = Screen(stdscr)   # create Screen object
-    stdscr.refresh()
-    ircFactory = IRCFactory(screen)
-    reactor.addReader(screen) # add screen object as a reader to the reactor
-    reactor.connectTCP("irc.freenode.net",6667,ircFactory) # connect to IRC
-    reactor.run() # have fun!
-    screen.close()
+    with Screen(stdscr) as screen:  # create Screen object
```

Comment:
Sounds good to me! Merge!

On Thursday, July 21, 2016, Craig Rodrigues <notifications@github.com>
wrote:

> In docs/words/examples/cursesclient.py
> <https://github.com/twisted/twisted/pull/377#discussion_r71811680>:
>
> >  if __name__ == '__main__':
> >      stdscr = curses.initscr() # initialize curses
> > -    screen = Screen(stdscr)   # create Screen object
> > -    stdscr.refresh()
> > -    ircFactory = IRCFactory(screen)
> > -    reactor.addReader(screen) # add screen object as a reader to the reactor
> > -    reactor.connectTCP("irc.freenode.net",6667,ircFactory) # connect to IRC
> > -    reactor.run() # have fun!
> > -    screen.close()
> > +    with Screen(stdscr) as screen:  # create Screen object
>
> I'm OK with using twisted.internet.task.react here.
>
> docs/words/examples/curesesclient.py needs more work, such as Python 3
> support
>
> import curses.wrapper
>
> fails on Python 3 for example.
>
> @markrwilliams <https://github.com/markrwilliams> if you want to submit a
> PR and fix up this example to use twisted.internet.task.react and also get
> it to work on Python 3, go for it. That would be great
>
> @scop <https://github.com/scop> I'm going to omit this file from the
> patchset and merge everything else in
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/twisted/twisted/pull/377/files/13eb01b44bb279f46042847c205f55f067f27ea3#r71811680>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq-99ekSSTU1kewJj5J0EEFpCea9VRdks5qYBB8gaJpZM4JRmzW>
> .
>